### PR TITLE
Fixed data tracking for Braze cards

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

The action cards on LLD were not registering the clicks and the flexible content cards on LLM always had the unique recipients at 0.

For LLD, the url property of the card had to be set for Braze to register the click (we set it to the card's id just so the value is not null).
For LLM, we had to log the impressions of the flexible cards so Braze could calculate the unique recipients.

Screenshot 2024-07-18 at 18 33 27
Screenshot 2024-07-18 at 18 33 40

### ❓ Context

- **JIRA or GitHub link**: [LIVE-12815] [LIVE-13258]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-12815]: https://ledgerhq.atlassian.net/browse/LIVE-12815?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/dynamicContent/ContentCardsCategory/Layout.tsx [hunk 1]
```diff
@@ -67,8 +67,16 @@ type LayoutProps = {
...
-  const { logClickCard, dismissCard, trackContentCardEvent } = useDynamicContent();
+  const { logClickCard, dismissCard, trackContentCardEvent, logImpressionCard } =
+    useDynamicContent();
+
+  useEffect(() => {
+    logImpressionCard(category.id);
+    for (const card of cards) {
+      logImpressionCard(card.id);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/dynamicContent/ContentCardsCategory/Layout.tsx:78]: Please do not "lie" to React: https://react.dev/learn/removing-effect-dependencies#why-is-suppressing-the-dependency-linter-so-dangerous

File: apps/ledger-live-desktop/src/renderer/hooks/useActionCards.tsx [hunk 2]
```diff
@@ -54,6 +54,11 @@ const useActionCards = () => {
     const actionCard = findActionCard(cardId);
 
     if (currentCard) {
+      // For some reason braze won't log the click event if the card url is empty
+      // Setting it as the card id just to have a dummy non empty value
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      currentCard.url = currentCard.id;
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/renderer/hooks/useActionCards.tsx:61]: Is it not dangerous to mutate the `currentCard` URL value? Don't we want to do it only for `logContentCardClick` function?

**Review by KVNLS — CHANGES_REQUESTED** [review]

File: apps/ledger-live-mobile/src/dynamicContent/ContentCardsCategory/Layout.tsx (hunk 1 above)
Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/dynamicContent/ContentCardsCategory/Layout.tsx:78]: Looking at the doc, while useEffectEvent isn't stable yet, I don't see how I can run a function at the component mount without having to mute the linter 🤔

File: apps/ledger-live-desktop/src/renderer/hooks/useActionCards.tsx (hunk 2 above)
Comment (cgrellard-ledger) [reply apps/ledger-live-desktop/src/renderer/hooks/useActionCards.tsx]: Technically this property is only used for "classic" content cards but as we only leverage the custom design ones this property is never set but ideally yes we shouldn't mutate an object like this. I pushed a modification :)

**Review by KVNLS — APPROVED** [review]

**Pull request merged**